### PR TITLE
feat: RFC commands

### PR DESCRIPTION
**Pull request by sustained**
# Todo

- [x] Add `RFCService`.
- [x] Add `!rfc` lookup command.
  - [x] Look up RFCs by:
    - [x] number
    - [x] title
    - [x] body
    - [x] author
    - [x] label(s)
- [x] Add `!rfcs` listing command.
  - [x] List RFCs by:
    - [x] all
    - [x] open
    - [x] closed
    - [ ] ~~popular~~ too painful to implement, abandoned
    - [x] merged
- [x] Add `!reload-rfcs` command for devs/staff.
- Improvements
  - [x] Use the new pagination system where applicable.
  - [x] Make everything look more like the `!api` embeds.

# Note

~~Relies on #25.~~

### CODE REVIEW

**Comment by sustained** [comment]
Ready for review.

File: src/commands/rfcs/reload.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        .setColor('GREEN')
+    } catch (error) {
+      console.error(error)
+      embed
+        .setDescription(
+          '❎ An error occured while fetching and recaching the RFC PRs.'
+        )
+        .setColor('RED')
+    } finally {
+      msg.channel.send(EMPTY_MESSAGE, { embed }).then(res => {
```
Comment (laquasicinque) [inline src/commands/rfcs/reload.js]: await?

File: src/commands/rfcs/rfc.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      success = true // For finally block.
+    } catch (error) {
+      if (error instanceof RFCDoesNotExistError) {
+        embed.setDescription('RFC not found!')
+      } else {
+        console.error(error)
+        embed.setDescription('Sorry, an unspecified error occured')
+      }
+    } finally {
+      msg.channel.send(EMPTY_MESSAGE, embed).then(reply => {
```
Comment (laquasicinque) [inline src/commands/rfcs/rfc.js]: await?

File: src/commands/rfcs/rfc.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+    return embed
+  }
+
+  /* eslint-disable no-unused-vars */
+  buildEmbedMultiple(embed, rfcs, author, filter, value) {
+    embed.setTitle(`Found Multiple Matching RFCs`)
+
+    // TODO: Could exceed max allowed number of fields (25)?
+    for (const rfc of rfcs) {
```
Comment (laquasicinque) [inline src/commands/rfcs/rfc.js]: could slice the rfcs?

Comment (sustained) [reply src/commands/rfcs/rfc.js]: I was just going to wait for #24 for this, personally.

File: src/commands/rfcs/reload.js (hunk 1 above)
Comment (sustained) [reply src/commands/rfcs/reload.js]: Fixed!

File: src/commands/rfcs/rfc.js (hunk 2 above)
Comment (sustained) [reply src/commands/rfcs/rfc.js]: Fixed!

**Pull request merged**